l10n_it_asset_management/migrations/16.0.1.0.0/pre-migration.py

```diff
@@ -0,0 +1,12 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def migrate(cr, installed_version):
+    if not installed_version or openupgrade.table_exists(
+        cr, "l10n_it_asset_management"
+    ):
```

Comment:
Dove verrebbe creata la tabella con lo stesso nome del modulo?

> aggiungere rename del modulo tramite opportuno hook / script di migrazione contestuale al commit in cui il modulo viene rinominato

Quello che intendeva @tafaRU in https://github.com/OCA/l10n-italy/pull/3271#pullrequestreview-1615593809 forse è una cosa tipo https://github.com/OCA/l10n-italy/commit/1194fb209a8f97410cbe780251de72637dde7d0d, la struttura utilizzata è riassunta anche in https://github.com/OCA/l10n-italy/issues/3574#issuecomment-1719725803.

> _**Come ho già scritto anche su Discord non riesco a testarlo per errori in OpenUpgrade, perciò finchè qualcuno riesce a provarlo non è da mergiare anche se gli altri test dovessero andare a buon fine.**_

Certo OpenUpgrade non è ancora pronto, ma cerchiamo di aggiungere lo stesso una traccia di migrazione che sarà poi probabilmente da correggere quando la si potrà provare.

Sicuramente non aspettiamo che OpenUpgrade sia pronto per poter mergiare le PR, anzi credo che si dovrebbe poter mergiare anche senza script di migrazione, e poi vengono aggiunti dopo, o almeno così avevo capito dalle varie chiamate e code sprint che abbiamo fatto.
Che ne dite @tafaRU @TheMule71?

Piuttosto @odooNextev, ricorda per favore di chiedere l'aggiornamento della revisione (come descritto in https://github.com/OCA/l10n-italy/pull/3424#discussion_r1273612829) quando hai finito con le modifiche perché altrimenti è difficile capire in che stato sia la PR.
Ad esempio: l'errore segnalato in https://github.com/OCA/l10n-italy/pull/3271#pullrequestreview-1482831360 è risolto?

Reply:
> Dove verrebbe creata la tabella con lo stesso nome del modulo?
> 
Non ricordo da dove avevo preso spunto, ma per esempio qui fa il controllo che non ci sia già la tabella e poi fa la rinomina, non so se serve anche rinominare qualche modello.
https://github.com/OCA/server-ux/blob/16.0/server_action_mass_edit/migrations/16.0.2.0.0/pre-migration.py
Forse vedendo questo esempio dovrei mettere il decoratore e prendere in ingresso `env`, poi ricavare `env.cr`.

> Piuttosto @odooNextev, ricorda per favore di chiedere l'aggiornamento della revisione (come descritto in [#3424 (comment)](https://github.com/OCA/l10n-italy/pull/3424#discussion_r1273612829)) quando hai finito con le modifiche perché altrimenti è difficile capire in che stato sia la PR. 
Non avevo richiesto l'aggiornamento perchè pensavo di riuscire a cavarmela con OpenUpgrade e quindi di terminarla, invece allora è un problema diverso.

> Ad esempio: l'errore segnalato in [#3271 (review)](https://github.com/OCA/l10n-italy/pull/3271#pullrequestreview-1482831360) è risolto?
Sì, a parte quello della migrazione non ci dovrebbero essere problemi.

Reply:
@SirAionTech @tafaRU con questo commit dovrei aver sistemato lo script di migrazione come chiedevate: https://github.com/OCA/l10n-italy/pull/3271/commits/73f64cfc4a422d057167b726774e6ddc7582bb9e

Reply:
Grazie, puoi mettere tutto nel commit di migrazione?
Altrimenti puoi anche lasciarlo in un commit dedicato ma rimuoverei le modifiche a `migrations` nel commit di migrazione.

Reply:
?